components/logins/src/migrate_sqlcipher_db.rs

```diff
@@ -5,11 +5,47 @@
 // Code to migrate from an sqlcipher DB to a plaintext DB
 
 use crate::db::MigrationMetrics;
+use crate::encryption::EncryptorDecryptor;
 use crate::error::*;
-use crate::schema;
+use crate::Login;
+use crate::LoginStore;
+use lazy_static::lazy_static;
 use rusqlite::{Connection, NO_PARAMS};
 use sql_support::ConnExt;
 use std::path::Path;
+use std::time::Instant;
+
+// Keep it around to be easy to reference the old schema
+pub const SQL_CIPHER_COLS: &str = "
+    guid,
+    username,
+    password,
+    hostname,
+    httpRealm,
+    formSubmitURL,
+    usernameField,
+    passwordField,
+    timeCreated,
+    timeLastUsed,
+    timePasswordChanged,
+    timesUsed
+";
+
+lazy_static! {
+    pub static ref GET_BY_GUID_SQL: String = format!(
+        "SELECT {common_cols}
+         FROM loginsL
+         WHERE is_deleted = 0
+
+         UNION ALL
+    
+         SELECT {common_cols}
+         FROM loginsM
+         WHERE is_overridden IS NOT 1
```

Comment:
I think you might need to remove the WHERE clauses and loop over each row from both tables to handle syncing correctly.  Some examples:

  - If there's a row in both tables, then loginsM contains the last known record from the server and loginsL contains the overridden value.  We want to copy both rows to make the merging logic correct.
  - If there's row in LoginsL with is_deleted=0, then I think that means the user deleted the login on this device.  We want to copy that row so that when we sync we treat that as a newly deleted login, rather than a newly added login from another device.

The way the schema is setup is pretty confusing, especially since it's different than most other components.  But I think copying/converting all the rows for both tables should work.

Reply:
I see, that indeed is a little confusing. I lifted this SQL to from the [db.rs](https://github.com/mozilla/application-services/blob/b43d90441d7dd160341cb22d8817aa146be50edb/components/logins/src/db.rs#L909) to closely match what we're currently doing in code.

But it seems like we might need to get those rows as well to ensure sync knows how to properly work with the tables